third_party/move/move-compiler-v2/src/pipeline/livevar_analysis_processor.rs

```diff
@@ -141,12 +155,58 @@ impl LiveVarAnalysisProcessor {
             code,
             &cfg,
         );
-        analyzer.state_per_instruction(state_map, code, &cfg, |before, after| {
-            LiveVarInfoAtCodeOffset {
-                before: before.livevars.clone().into_iter().collect(),
-                after: after.livevars.clone().into_iter().collect(),
+        // Prepare the result as a map from CodeOffset to LiveVarInfo
+        let mut code_map =
+            analyzer.state_per_instruction(state_map, code, &cfg, |before, after| {
+                LiveVarInfoAtCodeOffset {
+                    before: before.livevars.clone().into_iter().collect(),
+                    after: after.livevars.clone().into_iter().collect(),
+                }
+            });
+
+        // Now propagate to all branches in the code the after set of the branch instruction. Consider code as follows:
+        // ```
+        // L0: if c goto L1 else L2
+        // L1: ..
+        //     goto L0
+        // L2: ..
+        // ```
+        // The backwards analysis will not populate the before state of L2, which should be the same as the after
+        // state of L0. However, via the loop the after state of L0 will be incrementally grow (without effecting
+        // the before state of L2). The conclusion is that whatever the after state of L0 is, should be joined
+        // (union) to the before state of each branch target.
```

Comment:
This sentence seems wrong.  

I think if you use some more formal notation similar to the dataflow equations in Wikipedia things might be clearer.  Why does our analysis require forward flow from L0 to L2?

Reply:
There is now better documentation, I think the mentioning of fixpoint led to some confusion. This propagation is not part of the analysis, but of summarizing it.